### DISCUSSION TITLE
Invalid alpha validation.

### DISCUSSION THREAD

**Discussion by hmingv** [laravel/framework#42894] [Q&A] [answered]
- Laravel Version: 9.x

### Description:

https://laravel.com/docs/9.x/validation#rule-alpha

May I ask alphabetic characters. Does the meaning of this sentence include other languages, or is it just 26 English letters?

### Steps To Reproduce:

Use alpha for form validation, enter Chinese, the validation rule will pass

**Comment by hmingv** [comment]
This issue is related to this validation rule `preg_match('/^[\pL\pM]+$/u', $value);`

In `src/Illuminate/Validation/Concerns/ValidatesAttributes.php` file line 317

**Comment by joelclermont** [comment]
This regex is using [Unicode character properties](https://en.wikipedia.org/wiki/Unicode_character_property). It is many, many more characters than A-Z.

If you want something more restrictive, you could use the `regex` rule with exactly which ranges of letters you want to allow.

**Comment by hmingv** [comment]
Thanks @joelclermont

I know this regex match is a Unicode character. But I am confused about this document. My understanding of the document description is that it will verify whether it is an English letter, not a Unicode character.

[alpha](https://laravel.com/docs/9.x/validation#rule-alpha)
The field under validation must be entirely alphabetic characters.

**Comment by hmingv** [comment, marked as answer]
https://github.com/laravel/framework/discussions/45660

Change to a new link to discuss.
